cmd/osbuild-worker/main.go

```diff
@@ -84,6 +90,80 @@ func WatchJob(ctx context.Context, job worker.Job) {
 	}
 }
 
+// protect an AWS instance from scaling and/or terminating.
+func setProtection(protected bool) {
+	// create a new session
+	awsSession, err := session.NewSession()
+	if err != nil {
+		logrus.Debug("Error getting an AWS session")
+		return
+	}
+
+	// get the identity for the instanceID
+	identity, err := ec2metadata.New(awsSession).GetInstanceIdentityDocument()
+	if err != nil {
+		logrus.Debug("Error getting the identity document")
+		return
+	}
+
+	svc := autoscaling.New(awsSession)
+
+	// get the autoscaling group info for the auto scaling group name
+	asInstanceInput := &autoscaling.DescribeAutoScalingInstancesInput{
+		InstanceIds: []*string{
+			aws.String(identity.InstanceID),
+		},
+	}
+	asInstanceOutput, err := svc.DescribeAutoScalingInstances(asInstanceInput)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			logrus.Warningf("Error getting the Autoscaling instances: %s %s", aerr.Code(), aerr.Error())
+		} else {
+			logrus.Error("Error getting the Autoscaling instances: unknown")
```

Comment:
We should also print `err` here, otherwise we have no clue what actually failed.

Reply:
(I'm actually not sure if we actually need to distinguish between the two error types, I had a brief look and it seems like `aerr.Error()` always includes `aerr.Code()`. Up to you, no hard feelings.